### PR TITLE
Add comments on ImageMagick requirement

### PR DESCRIPTION
**Pull request by mxd4**
Subject: Add comments  to clarify the cause of issue #10384 

### Purpose
Allow newcomers to quickly understand the requirement to ImageMagick if `test_ext_imgconverter.py` fails.

NB: First contribution, thanks for reviewing! :)

### CODE REVIEW

File: tests/test_ext_imgconverter.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
-"""Test sphinx.ext.imgconverter extension."""
+"""Test sphinx.ext.imgconverter extension.
+Requires ImageMagick to be installed, otherwise the conversion will be skipped and this test will fail.
```
Comment (AA-Turner) [inline tests/test_ext_imgconverter.py]:
```suggestion

Unix only. Fails if ImageMagick_ is not installed.

.. _ImageMagick: https://imagemagick.org
```

File: doc/usage/extensions/imgconverter.rst [hunk 2]
```diff
@@ -12,7 +12,8 @@ This extension converts images in your document to appropriate format for
 builders.  For example, it allows you to use SVG images with LaTeX builder.
 As a result, you don't mind what image format the builder supports.
 
-Internally, this extension uses Imagemagick_ to convert images.
+Internally, this extension uses Imagemagick_ to convert images,
+you'll have to install it on your machine.
 
 .. _Imagemagick: https://www.imagemagick.org/script/index.php
```
Comment (AA-Turner) [inline doc/usage/extensions/imgconverter.rst]:
```suggestion
.. _ImageMagick: https://www.imagemagick.org
```

File: doc/usage/extensions/imgconverter.rst [hunk 3]
```diff
@@ -12,7 +12,8 @@ This extension converts images in your document to appropriate format for
 builders.  For example, it allows you to use SVG images with LaTeX builder.
 As a result, you don't mind what image format the builder supports.
 
-Internally, this extension uses Imagemagick_ to convert images.
+Internally, this extension uses Imagemagick_ to convert images,
+you'll have to install it on your machine.
```
Comment (AA-Turner) [inline doc/usage/extensions/imgconverter.rst]:
```suggestion
The extension uses ImageMagick_ to perform conversions, 
and will fail if ImageMagick is not installed.
```

**Review by AA-Turner — CHANGES_REQUESTED** [review]

**Comment by mxd4** [comment]
CI [failed](https://github.com/sphinx-doc/sphinx/actions/runs/2215634837) on mypy and docslint steps.
I fixed the docslint error and rebased `5.x` to get the mypy fix.
@tk0miya could you run it again? 🙏🏻

**Comment by mxd4** [comment]
Thanks a lot for the suggestions @AA-Turner
All done :)

**Comment by AA-Turner** [comment]
I think more usefully you should also change the warning

https://github.com/sphinx-doc/sphinx/blob/8c7a472d58f832b9e2bd05d9d0fd5dbe0fcf077f/sphinx/ext/imgconverter.py#L32-L36

Something like the following may be more useful:

```diff
-            logger.warning(__('convert command %r cannot be run, '
-                              'check the image_converter setting: %s'),
-                           self.config.image_converter, exc)
+            logger.warning(__(
+                'Unable to run the image conversion command {self.config.image_converter!r}! '
+                'sphinx.ext.imgconverter' requires ImageMagick by defult. '
+                'Ensure it is installed, or set the configuration option '
+                "'image_converter' to a custom conversion command.\n\n"
+                'Traceback: {exc}'
+            ))
```

A

**Comment by AA-Turner** [comment]
I'm somewhat conflicted on the change of the documentation -- as noted in the extension, ImageMagick is just a default, and can be changed. Perhaps add "by default" or similar?

https://github.com/sphinx-doc/sphinx/blob/8c7a472d58f832b9e2bd05d9d0fd5dbe0fcf077f/sphinx/ext/imgconverter.py#L69-L79

A

**Comment by mxd4** [comment]
@AA-Turner I agree. 
I've just added the suggested warning, as well as changes to the doc to make it clearer.
It's now explicit that the provided command is `convert` by default and uses ImageMagick, but can be overridden.

File: doc/usage/extensions/imgconverter.rst [hunk 4]
```diff
@@ -31,10 +32,12 @@ Configuration
 
 .. confval:: image_converter
 
-   A path to :command:`convert` command.  By default, the imgconverter uses
+   A path to a conversion command.  By default, the imgconverter finds
    the command from search paths.
 
-   On windows platform, :command:`magick` command is used by default.
+   On Unix platform, :command:`convert` command is used by default.
```
Comment (AA-Turner) [inline doc/usage/extensions/imgconverter.rst]:
```suggestion
   On Unix platforms, the command :command:`convert` is used by default.
```

File: doc/usage/extensions/imgconverter.rst [hunk 5]
```diff
@@ -31,10 +32,12 @@ Configuration
...
+
+   On Windows platform, :command:`magick` command is used by default.
```
Comment (AA-Turner) [inline doc/usage/extensions/imgconverter.rst]:
```suggestion
   On Windows, the command :command:`magick` is used by default.
```

File: doc/usage/extensions/imgconverter.rst [hunk 6]
```diff
@@ -45,8 +48,8 @@ Configuration
    Additional command-line arguments to give to :command:`convert`, as a list.
    The default is an empty list ``[]``.
 
-   On windows platform, it defaults to ``["convert"]``.
+   On Windows platform, it defaults to ``["convert"]``.
```
Comment (AA-Turner) [inline doc/usage/extensions/imgconverter.rst]:
```suggestion
   On Windows, it defaults to ``["convert"]``.
```

**Review by AA-Turner — APPROVED** [review]
Grammar suggestions, otherwise seems alright!

A

**Comment by tk0miya** [comment]
Thank you for your work, and sorry for the incovinience. I think it's better to skip the test if ImageMagick not found. So I just posted https://github.com/sphinx-doc/sphinx/pull/10411 to do that.

File: sphinx/ext/imgconverter.py [hunk 7]
```diff
@@ -30,9 +30,13 @@ def is_available(self) -> bool:
             subprocess.run(args, stdout=PIPE, stderr=PIPE, check=True)
             return True
         except OSError as exc:
-            logger.warning(__('convert command %r cannot be run, '
-                              'check the image_converter setting: %s'),
-                           self.config.image_converter, exc)
+            logger.warning(__(
+                f"Unable to run the image conversion command {self.config.image_converter!r}!"
+                "\n'sphinx.ext.imgconverter' requires ImageMagick by default. "
```
Comment (tk0miya) [inline sphinx/ext/imgconverter.py]: Is there any reason to put "\n" on the top of the line?

File: sphinx/ext/imgconverter.py [hunk 8]
```diff
@@ -30,9 +30,13 @@ def is_available(self) -> bool:
             subprocess.run(args, stdout=PIPE, stderr=PIPE, check=True)
             return True
         except OSError as exc:
-            logger.warning(__('convert command %r cannot be run, '
-                              'check the image_converter setting: %s'),
-                           self.config.image_converter, exc)
+            logger.warning(__(
+                f"Unable to run the image conversion command {self.config.image_converter!r}!"
```
Comment (tk0miya) [inline sphinx/ext/imgconverter.py]: Usually, we don't use exclamation marks for warning. Could you replace it with a period?

File: tests/test_ext_imgconverter.py [hunk 9]
```diff
@@ -1,4 +1,9 @@
-"""Test sphinx.ext.imgconverter extension."""
+"""Test sphinx.ext.imgconverter extension.
+
+Unix only. Fails if ImageMagick_ is not installed.
```
Comment (tk0miya) [inline tests/test_ext_imgconverter.py]: I believe https://github.com/sphinx-doc/sphinx/pull/10411 will be able to remove this comment.

**Review by tk0miya — CHANGES_REQUESTED** [review]
LGTM with nits.

File: sphinx/ext/imgconverter.py (hunk 7 above)
Comment (mxd4) [reply sphinx/ext/imgconverter.py]: I wanted a bit more readability, first I added it at the end of line 34 but it made the line too long for flake8.
However if it's bad practice to put `\n` in the middle of warning messages I can remove it.

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Sorry for my late response. Merging now. Thank you for your contribution!

**Pull request merged**